airqo-platform/AirQo-api
# Events monitor

# Events monitor

**_WHAT DOES THIS PR DO?_**

- [x] Performs periodic get requests to the [get events endpoint](https://docs.airqo.net/airqo-platform-api/device-registry#get-events), checks number of events returned, status code and date of last event
- [x] Notifies slack incase the returned status code is not 200 or events array is empty

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
None

**_WHAT IS THE LINK TO THE PR BRANCH_**
[events-monitor](https://github.com/airqo-platform/AirQo-api/tree/events-monitor)

**_HOW DO I TEST OUT THIS PR?_**
Make sure you have a latest [.env](https://docs.google.com/document/d/1NOlvUhNnigagqARtLgfyIlsCDWtXaiGhns77kiFJUTU/edit)
```
cd AirQo-api/src/data-mgt/python/events-monitor/
python3 -m venv venv
source venv/bin/activate
pip install -r requirements.txt
python main.py
```
You should receive a slack notification

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
None

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
None

**_ARE THERE ANY RELATED PRs?_**
None
